api/v3/FinancialTrxn.php

```diff
@@ -39,7 +39,11 @@
  * @return array
  */
 function civicrm_api3_financial_trxn_create($params) {
-  return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (empty($params['id']) && empty($params['contribution_id']) && empty($params['entity_id'])) {
```

Comment:
@eileenmcnaughton actually ```civicrm_api3_verify_mandatory(...)``` perform OR operation between keys but here we need AND. I changed it to this.